File: golang/1_String_Reversal/solution2.go
package main

import "fmt"

func main() {
	fmt.Println(reverse2("APPLE"))
}

func reverse2(s string) (result string) {
	for _, char := range s {
		fmt.Println(string(char))
		result = string(char) + result
	}
	return
}

File: python/1_String_Reversal/solution2.py
# Reverse a string using recursion
def reverse2(s):
    if len(s) == 0:
        return s
    else:
        print(s[1:])
        print(s[0])
        return reverse2(s[1:])+s[0]


s = "APPLE"
print(reverse2(s))
